### PR TITLE
wxGUI/Single-Window: New change page event for AuiNotebook

### PR DESCRIPTION
**Pull request by lindakladivova**

### CODE REVIEW

File: gui/wxpython/main_window/frame.py [hunk 1]
```diff
@@ -309,6 +309,10 @@ def _createMapNotebook(self):
             agwStyle=notebook_style,
         )
         self.mapnotebook.SetArtProvider(aui.AuiDefaultTabArt())
+        # bindings
+        self.mapnotebook.Bind(
+            aui.EVT_AUINOTEBOOK_PAGE_CHANGED, self.OnMapNotebookPageChanged
```
Comment (petrasovaa) [inline gui/wxpython/main_window/frame.py:314]: There is _onMapDisplayFocus in lmgr/frame.py, which seems to be doing what you need, why this new method?

File: gui/wxpython/main_window/frame.py [hunk 2]
```diff
@@ -912,6 +917,24 @@ def OnCBPageClosing(self, event):
...
+            return
+
+        try:
+            currentMapDisp = self.mapnotebook.GetCurrentPage()
+            pgnum = currentMapDisp.getLayerTreeIndex()
+            if pgnum > -1:
+                self.notebookLayers.SetSelection(pgnum)
+                self.currentPage = self.notebookLayers.GetCurrentPage()
+                self.currentPageNum = self.notebookLayers.GetSelection()
+        except Exception:
```
Comment (petrasovaa) [inline gui/wxpython/main_window/frame.py:933]: What exception is actually thrown here? It's better to use the specific one to make it more clear what it is trying to do.

Comment (lindakladivova) [reply gui/wxpython/main_window/frame.py:933]: It is TypeError. As I understand it right, the OnMapNotebookChanged is also called during the processing of NewDisplay method. The map panel is already known but the layertree not, so thats the reason why it throws this exception.

**Review by petrasovaa — APPROVED** [review]

**Review by petrasovaa — APPROVED** [review]
Looks good!

**Pull request merged**